onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2674,6 +2672,282 @@ Status LRNOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
+    return false;
+  }
+
+  float min = std::numeric_limits<float>::lowest();
+  float max = std::numeric_limits<float>::max();
+  if (!GetMinMax(model_builder, node, min, max))
+    return false;
+
+  // We only supoort relu6 or relu1
+  // TODO, support clip between 2 arbitrary numbers
+  if ((min == 0.0f && max == 6.0f) || (min == -1.0f && max == 1.0f)) {
+    return true;
+  } else {
+    LOGS_DEFAULT(VERBOSE) << "Clip only supports [min, max] = [0, 6] or [-1, 1], the input is ["
+                          << min << ", " << max << "]";
+    return false;
+  }
+
+  return true;
+}
+
+Status ClipOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node) {
+  auto& shaper(model_builder.GetShaper());
+  const auto& operand_indices(model_builder.GetOperandIndices());
+  const auto& operand_types(model_builder.GetOperandTypes());
+
+  const auto& input = node.InputDefs()[0]->Name();
+  const auto& output = node.OutputDefs()[0]->Name();
+  bool output_is_nhwc = model_builder.IsOperandNHWC(input);
+
+  ORT_RETURN_IF_ERROR(shaper.Identity(input, output));
+  const OperandType output_operand_type(operand_types.at(input).type, shaper[output]);
+
+  float min = std::numeric_limits<float>::lowest();
+  float max = std::numeric_limits<float>::max();
+  GetMinMax(model_builder, node, min, max);
+
+  int32_t op_code;
+  if (min == 0.0f && max == 6.0f)
+    op_code = ANEURALNETWORKS_RELU6;
+  else if (min == -1.0f && max == 1.0f)
+    op_code = ANEURALNETWORKS_RELU1;
+  else
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "ClipOpBuilder, unsupported input [", min, ", ", max, "]");
+
+  std::vector<uint32_t> input_indices;
+  input_indices.push_back(operand_indices.at(input));
+  ORT_RETURN_IF_ERROR(model_builder.AddOperation(op_code, input_indices,
+                                                 {output}, {output_operand_type}, {output_is_nhwc}));
+  return Status::OK();
+}
+
+#pragma endregion
+
+#pragma region op_Resize
+
+class ResizeOpBuilder : public BaseOpBuilder {
+ public:
+  void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) override;
+
+ private:
+  bool IsOpSupportedImpl(ModelBuilder& model_builder, const Node& node) override;
+
+  int32_t GetMinSupportedSdkVer(ModelBuilder& /* model_builder */, const Node& node) const override;
+
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node) override ORT_MUST_USE_RESULT;
+};
+
+int32_t ResizeOpBuilder::GetMinSupportedSdkVer(ModelBuilder& /* model_builder */, const Node& node) const {
+  NodeAttrHelper helper(node);
+  if (helper.Get("mode", "nearest") == "nearest")
+    return 29;
+
+  return 28;
+}
+
+void ResizeOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) {
+  model_builder.AddInitializerToSkip(node.InputDefs()[2]->Name());  // scales
+
+  if (node.InputDefs().size() > 3)
+    model_builder.AddInitializerToSkip(node.InputDefs()[3]->Name());  // sizes
+}
```

Comment:
Should be one or the other as the spec says "Only one of 'scales' and 'sizes' can be specified."

Reply:
We will still need to add  scales to the skipped initializers, since currently we do not handle scalar initializers, update comments

---
In reply to: [502978465](https://github.com/microsoft/onnxruntime/pull/5427#discussion_r502978465) [](ancestors = 502978465)